src/pds/registrysweepers/utils/__init__.py

```diff
@@ -0,0 +1,214 @@
+    )
+    ap.add_argument("-b", "--base-URL", required=True, type=str)
+    ap.add_argument(
+        "-c",
+        "--ccs-remotes",
+        default=[],
+        nargs="*",
+        help="names of additional opensearch cross-cluster remotes, space-separated",
+    )
+    ap.add_argument("-l", "--log-file", default=None, required=False, help="file to write the log messages")
+    ap.add_argument(
+        "-L",
+        "--log-level",
+        default="ERROR",
+        required=False,
+        type=parse_log_level,
+        help="Python logging level as an int or string like INFO for logging.INFO [%(default)s]",
+    )
+    ap.add_argument(
+        "-p",
+        "--password",
+        default=None,
+        required=False,
+        help="password to login to opensearch leaving it blank if opensearch does not require login",
+    )
+    ap.add_argument(
+        "-u",
+        "--username",
+        default=None,
+        required=False,
+        help="username to login to opensearch leaving it blank if opensearch does not require login",
+    )
+    ap.add_argument("-v", "--verify", action="store_true", default=False, help="verify the host certificates")
+
+    args = ap.parse_args()
+    return args
+
+
+def parse_log_level(input: str) -> int:
+    """Given a numeric or uppercase descriptive log level, return the associated int"""
+    try:
+        result = int(input)
+    except ValueError:
+        result = getattr(logging, input)
+    return result
+
+
+def _vid_as_tuple_of_int(lidvid: str):
+    major_version, minor_version = lidvid.split("::")[1].split(".")
+    return (int(major_version), int(minor_version))
+
+
+def configure_logging(filepath: Union[str, None], log_level: int):
+    logging.root.handlers = []
+    handlers: List[logging.StreamHandler] = [logging.StreamHandler()]
+
+    if filepath:
+        handlers.append(logging.FileHandler(filepath))
+
+    logging.basicConfig(level=log_level, format="%(asctime)s::%(levelname)s::%(message)s", handlers=handlers)
+
+
+def query_registry_db(
+    host: HOST,
+    query: Dict,
+    _source: Dict,
+    index_name: str = "registry",
+    page_size: int = 10000,
+    scroll_validity_duration_minutes: int = 10,
+) -> Iterable[Dict]:
+    """
+    Given an OpenSearch host and query/_source, return an iterable collection of hits
+
+    Example query: {"bool": {"must": [{"terms": {"ops:Tracking_Meta/ops:archive_status": ["archived", "certified"]}}]}}
+    Example _source: {"includes": ["lidvid"]}
+    """
+
+    req_content = {
+        "query": query,
+        "_source": _source,
+        "size": page_size,
+    }
+
+    log.debug(f"Initiating query: {req_content}")
+
+    cross_cluster_indexes = [f"{node}:{index_name}" for node in host.cross_cluster_remotes]
+    path = ",".join([index_name] + cross_cluster_indexes) + f"/_search?scroll={scroll_validity_duration_minutes}m"
+    served_hits = 0
+
+    more_data_exists = True
+    while more_data_exists:
+        resp = requests.get(
```

Comment:
Curious why we're using Python Requests instead of `opensearch-py`, the OpenSearch-branded Python client. Is it to support mocked servers?

Reply:
Because provenance.py [used to be a script I didn't understand well enough to touch](https://github.com/NASA-PDS/registry-api/commit/f8ec2e38ac7f592afe03dbf3967a123047819ca4#diff-29c43a69d7f3bda3a91776c36442e1b6fcf9113062929423cf71b93b9f940275) and then it was [broken apart incrementally](https://github.com/NASA-PDS/registry-api/commit/1ed17117cd6e41b33e2f7d02e49d208147784182) and never thought to ask myself whether a python-client existed before I extracted reusable functions from provenance.py to a utility library.

I'll open a ticket to switch over to that, since it's important but not blocking.